pynest/examples/EI_clustered_network/network.py

```diff
@@ -0,0 +1,807 @@
+        pop_size_I = self._params["N_I"] // self._params["n_clusters"]
+        E_pops = [
+            nest.Create(self._params["neuron_type"], n=pop_size_E, params=E_neuron_params)
+            for _ in range(self._params["n_clusters"])
+        ]
+        I_pops = [
+            nest.Create(self._params["neuron_type"], n=pop_size_I, params=I_neuron_params)
+            for _ in range(self._params["n_clusters"])
+        ]
+
+        self._populations = [E_pops, I_pops]
+
+    def connect(self):
+        """Connect the excitatory and inhibitory populations with each other
+        in the EI-clustered scheme
+
+        Raises
+        ------
+        ValueError
+            If the clustering method is not recognized
+        """
+
+        if "clustering" not in self._params or self._params["clustering"] == "weight":
+            self.connect_weight()
+        elif self._params["clustering"] == "probabilities":
+            self.connect_probabilities()
+        else:
+            raise ValueError("Clustering method %s not implemented" % self._params["clustering"])
+
+    def connect_probabilities(self):
+        """Connect the clusters with a probability EI-clustered scheme
+
+        Connects the excitatory and inhibitory populations with each other
+        in the EI-clustered scheme by increasing the probabilities of the
+        connections within the clusters and decreasing the probabilities of the
+        connections between the clusters. The weights are calculated so that
+        the total input to a neuron is balanced.
+        """
+        #  self._populations[0] -> Excitatory population
+        #  self._populations[1] -> Inhibitory population
+        # connectivity parameters
+        js = self._params["js"]  # connection weights
+        N = self._params["N_E"] + self._params["N_I"]  # total units
+        if np.isnan(js).any():
+            js = helper.calculate_RBN_weights(self._params)
+        js *= self._params["s"]
+
+        if self._params["n_clusters"] > 1:
+            pminus = (self._params["n_clusters"] - self._params["pplus"]) / float(self._params["n_clusters"] - 1)
+        else:
+            self._params["pplus"] = np.ones((2, 2))
+            pminus = np.ones((2, 2))
+
+        p_plus = self._params["pplus"] * self._params["baseline_conn_prob"]
+        p_minus = pminus * self._params["baseline_conn_prob"]
+
+        iterations = np.ones((2, 2), dtype=int)
+        # test if any of the probabilities is larger than 1
+        if np.any(p_plus > 1):
+            print("The probability of connection is larger than 1")
+            print("p_plus: ", p_plus)
+            print("p_minus: ", p_minus)
+            for i in range(2):
+                for j in range(2):
+                    if p_plus[i, j] > 1:
+                        iterations[i, j] = int(np.ceil(p_plus[i, j]))
+                        p_plus[i, j] /= iterations[i, j]
+            print("p_plus: ", p_plus)
+            print(iterations)
+
```

Comment:
Clarify here that this is the result of the iterative adjustment and that the iteration is now complete. Use format strings for pretty output (sensible number of decimals), see the suggestions for the output firing rates.